src/services/textChanges.ts

```diff
@@ -161,6 +165,10 @@ namespace ts.textChanges {
             return new ChangeTracker(context.newLineCharacter === "\n" ? NewLineKind.LineFeed : NewLineKind.CarriageReturnLineFeed, context.rulesProvider);
         }
 
+        public static fromRefactorContext(context: RefactorContext) {
+            return new ChangeTracker(context.newLineCharacter === "\n" ? NewLineKind.LineFeed : NewLineKind.CarriageReturnLineFeed, context.rulesProvider);

```

Comment:
This appears to be the third occurrence.  Consider extracting `context.newLineKind`.